packages/dev/core/src/Meshes/Compression/Draco/dracoCommons.ts

```diff
@@ -0,0 +1,220 @@
+    // indicates the presision of each type of data stored in the draco file
+    quantizationBits?: { [key: string]: number };
+    // indicate we should export normals if present
+    exportNormals?: boolean;
+    // indicate we should export tangents if present
+    exportTangents?: boolean;
+    // indicate we should export texture coordinates if present
+    exportUvs?: boolean;
+    // indicate we should export colors if present
+    exportColors?: boolean;
+    // indicate we should export weights if present
+    exportWeights?: boolean;
+}
+
+export interface IDracoEncoder {
+    encodeMeshAsync(input: IGetSetVerticesData, options?: IDracoEncoderOptions): Promise<Nullable<IDracoEncodedPrimitive>>;
+}
+
+export interface IDracoDecoder {
+    decodeMeshAsync(data: ArrayBuffer | ArrayBufferView, attributes?: { [kind: string]: number }, dividers?: { [kind: string]: number }): Promise<VertexData>;
+}
+
+/**
+ * Configuration for Draco codecs - either encoder or decoder
+ */
+export interface IDracoCompressionEngineConfiguration {
+    /**
+     * The url to the WebAssembly module.
+     */
+    wasmUrl?: string;
+
+    /**
+     * The url to the WebAssembly binary.
+     */
+    wasmBinaryUrl?: string;
+
+    /**
+     * The url to the fallback JavaScript module.
+     */
+    fallbackUrl?: string;
+}
+
+class DracoEngineInfo {
+    static FromConfiguration(config: IDracoCompressionEngineConfiguration): DracoEngineInfo {
+        return config.wasmUrl && config.wasmBinaryUrl && typeof WebAssembly === "object"
+            ? {
+                  url: Tools.GetAbsoluteUrl(config.wasmUrl),
+                  wasmBinaryPromise: Tools.LoadFileAsync(Tools.GetAbsoluteUrl(config.wasmBinaryUrl)),
+              }
+            : {
+                  url: Tools.GetAbsoluteUrl(config.fallbackUrl!),
+                  wasmBinaryPromise: Promise.resolve(undefined),
+              };
+    }
+
+    url: string | undefined;
+    wasmBinaryPromise: Promise<ArrayBuffer | string | undefined>;
+}
+
+/**
+ * This is the base class for the Draco Encoder and Decoder.
+ */
+export abstract class DracoCompressionBase implements IDisposable {
+    private static _GetDefaultNumWorkers(): number {
+        if (typeof navigator !== "object" || !navigator.hardwareConcurrency) {
+            return 1;
+        }
+
+        // Use 50% of the available logical processors but capped at 4.
+        return Math.min(Math.floor(navigator.hardwareConcurrency * 0.5), 4);
+    }
+
+    public static WasmBaseUrl: string = "https://preview.babylonjs.com/";
+
+    protected static _isCodecAvailable(codec: IDracoCompressionEngineConfiguration) {
+        return !!((codec.wasmUrl && codec.wasmBinaryUrl && typeof WebAssembly === "object") || codec.fallbackUrl);
+    }
+
+    /**
+     * Default number of workers to create when creating the draco compression object.
+     */
+    public static DefaultNumWorkers = DracoCompressionBase._GetDefaultNumWorkers();

```

Comment:
This should be a public property on the decoder and encoder. I don't think we want users to touch the base class to change this value.